--- G3D.lib/source/BinaryOutput.cpp	2010-04-19 15:45:10.000000000 -0400
+++ /Users/mixamolaptop3/Downloads/G3D/G3D.lib/source/BinaryOutput.cpp	2012-09-14 03:11:31.000000000 -0400
@@ -14,7 +14,11 @@
 #include "G3D/FileSystem.h"
 #include "G3D/stringutils.h"
 #include "G3D/Array.h"
-#include <zlib.h>
+
+#ifdef G3D_ENABLE_ZLIB
+#  include <zlib.h>
+#endif
+
 #include "G3D/Log.h"
 #include <cstring>
 
@@ -270,6 +274,7 @@
 
 
 void BinaryOutput::compress() {
+#ifdef G3D_ENABLE_ZLIB
     if (m_alreadyWritten > 0) {
         throw "Cannot compress huge files (part of this file has already been written to disk).";
     }
@@ -308,6 +313,9 @@
     m_pos = m_bufferLen;
 
     System::free(temp);
+#else
+    throw "No zlib support, can't compress";
+#endif
 }
 
 
