--- debugAssert.cpp.old	2010-08-07 23:59:57.000000000 -0700
+++ debugAssert.cpp	2010-08-08 00:01:27.000000000 -0700
@@ -37,8 +37,10 @@
 AssertionHook _failureHook = _handleErrorCheck_;
 
 #ifdef G3D_LINUX
+  #ifndef G3D_DEBUG_NOGUI
     Display*      x11Display = NULL;
     Window        x11Window  = 0;
+  #endif
 #endif
 
 
@@ -250,6 +252,7 @@
         ClipCursor(NULL);
         
     #elif defined(G3D_LINUX)
+      #ifndef G3D_DEBUG_NOGUI
         if (x11Display != NULL) {
             XUngrabPointer(x11Display, CurrentTime);
             XUngrabKeyboard(x11Display, CurrentTime);
@@ -264,6 +267,7 @@
             XAllowEvents(x11Display, AsyncPointer, CurrentTime);
             XFlush(x11Display);
         }
+      #endif
     #elif defined(G3D_OSX)
         // TODO: OS X
     #endif
