--- G3D.lib/source/GImage.cpp	2010-04-20 03:22:48.000000000 -0400
+++ /Users/mixamolaptop3/Downloads/G3D/G3D.lib/source/GImage.cpp	2012-09-14 03:11:31.000000000 -0400
@@ -17,10 +17,12 @@
 #include "G3D/Log.h"
 #include "G3D/fileutils.h"
 
+#ifdef G3D_ENABLE_ZLIB
 #ifdef G3D_LINUX
 #    include <png.h>
 #else
-#    include "png.h"
+#    include "../../png.lib/include/png.h"
+#endif
 #endif
 
 #include <sys/stat.h>
@@ -455,11 +457,13 @@
         return PPM_BINARY;
     }
 
+#ifdef G3D_ENABLE_ZLIB
     if (dataLen > 8) {
         if (!png_sig_cmp((png_bytep)data, 0, 8)) {
             return PNG;
         }
     }
+#endif
 
     if ((dataLen > 0) && (data[0] == 'B')) {
         return BMP;
