--- Matrix3.h.old	2010-09-22 22:52:52.000000000 -0700
+++ Matrix3.h	2010-09-22 22:53:25.000000000 -0700
@@ -209,8 +209,8 @@
     bool isOrthonormal() const;
 
     Matrix3 transpose () const;
-    bool inverse (Matrix3& rkInverse, float fTolerance = 1e-06) const;
-    Matrix3 inverse (float fTolerance = 1e-06) const;
+    bool inverse (Matrix3& rkInverse, float fTolerance = 1e-30) const;
+    Matrix3 inverse (float fTolerance = 1e-30) const;
     float determinant () const;
 
     /** singular value decomposition */
