--- G3D.lib/include/G3D/debugAssert.h	2010-01-27 12:19:49.000000000 -0500
+++ /Volumes/BOOTCAMP/Users/sid/Documents/GitHub/Fuse_OldWin/Code/Dependencies/Thea/Code/Dependencies/G3D/G3D//G3D.lib/include/G3D/debugAssert.h	2012-08-18 01:34:14.000000000 -0400
@@ -36,12 +36,17 @@
 #   pragma warning (disable : 4127)
 #endif
 
+// SC: We don't want to pull in an X dependency
+#define G3D_DEBUG_NOGUI
+
 #ifdef G3D_LINUX
+  #ifndef G3D_DEBUG_NOGUI
     // Needed so we can define a global display
     // pointer for debugAssert.
     #include <X11/Xlib.h>
     #include <X11/Xutil.h>
     #include <X11/Xatom.h>
+  #endif
 #endif
 
 
@@ -178,6 +183,7 @@
 namespace G3D {  namespace _internal {
 
 #ifdef G3D_LINUX
+  #ifndef G3D_DEBUG_NOGUI
     /**
      A pointer to the X11 display.  Initially NULL.  If set to a
      non-null value (e.g. by SDLWindow), debugAssert attempts to use
@@ -193,6 +199,7 @@
      fails.
      */
     extern Window        x11Window;
+  #endif
 #endif
 
 /**
