--- G3D.lib/include/G3D/Matrix3.h	2010-01-27 12:19:49.000000000 -0500
+++ ../../../../../../Downloads/G3D.lib/include/G3D/Matrix3.h	2012-08-18 01:34:14.000000000 -0400
@@ -209,8 +209,8 @@
     bool isOrthonormal() const;
 
     Matrix3 transpose () const;
-    bool inverse (Matrix3& rkInverse, float fTolerance = 1e-06) const;
-    Matrix3 inverse (float fTolerance = 1e-06) const;
+    bool inverse (Matrix3& rkInverse, float fTolerance = 1e-30) const;
+    Matrix3 inverse (float fTolerance = 1e-30) const;
     float determinant () const;
 
     /** singular value decomposition */
