--- G3D.lib/include/G3D/platform.h	2010-07-17 09:03:43.000000000 -0400
+++ /Users/mixamolaptop3/Downloads/G3D/G3D.lib/include/G3D/platform.h	2012-09-14 03:11:31.000000000 -0400
@@ -12,6 +12,11 @@
 #ifndef G3D_platform_h
 #define G3D_platform_h
 
+// Added by S.C., Aug 2012 to avoid duplicate symbol errors c.f. Qt when compiling on Win64.
+#if !defined(_MSC_VER)
+#  define G3D_ENABLE_ZLIB
+#endif
+
 /**
  The version number of G3D in the form: MmmBB -> 
  version M.mm [beta BB]
@@ -57,7 +62,9 @@
 #elif defined(__linux__)
     #define G3D_LINUX
 #elif defined(__APPLE__)
-    #define G3D_OSX
+    // For 64-bit OS X compilation, pretend this is Linux to eliminate all Carbon dependencies
+    #define  G3D_LINUX
+    // #define G3D_OSX
 
    // Prevent OS X fp.h header from being included; it defines
    // pi as a constant, which creates a conflict with G3D
