--- G3D.lib/include/G3D/stringutils.h	2010-03-05 22:26:03.000000000 -0500
+++ /Users/mixamolaptop3/Downloads/G3D/G3D.lib/include/G3D/stringutils.h	2012-09-14 03:11:31.000000000 -0400
@@ -32,27 +32,21 @@
   \sa G3D::findLastSlash, G3D::isSlash
 */
 inline int findSlash(const std::string& f, int start = 0) {
-    int i = f.find('/', start);
-    int j = f.find('\\', start);
-    if (((i != -1) && (i < j)) || (j == -1)) {
-        return i;
-    } else {
-        return j;
-    }
+    std::size_t pos = f.find_first_of("/\\", (std::size_t)start);
+    return pos == std::string::npos ? -1 : (int)pos;
 }
 
 
-/** Finds the index of the first '\\' or '/' character, starting at index \a start (if \a start is -1, starts at the end of the string).
+/** Finds the index of the last '\\' or '/' character, starting at index \a start (if \a start is -1, starts at the end of the string).
   \sa G3D::findSlash, G3D::isSlash
   */
 inline int findLastSlash(const std::string& f, int start = -1) {
     if (start == -1) {
-        start = f.length() - 1;
+        start = (int)f.length() - 1;
     }
 
-    int i = f.rfind('/', start);
-    int j = f.rfind('\\', start);
-    return max(i, j);
+    std::size_t pos = f.find_last_of("/\\", (std::size_t)start);
+    return pos == std::string::npos ? -1 : (int)pos;
 }
 
 /**
