--- G3D.lib/include/G3D/stringutils.h	2010-03-05 22:26:03.000000000 -0500
+++ ../../../../../../Downloads/G3D.lib/include/G3D/stringutils.h	2012-08-19 19:10:26.000000000 -0400
@@ -32,12 +32,12 @@
   \sa G3D::findLastSlash, G3D::isSlash
 */
 inline int findSlash(const std::string& f, int start = 0) {
-    int i = f.find('/', start);
-    int j = f.find('\\', start);
+    std::size_t i = f.find('/', start);
+    std::size_t j = f.find('\\', start);
     if (((i != -1) && (i < j)) || (j == -1)) {
-        return i;
+        return (int)i;
     } else {
-        return j;
+        return (int)j;
     }
 }
 
@@ -47,12 +47,12 @@
   */
 inline int findLastSlash(const std::string& f, int start = -1) {
     if (start == -1) {
-        start = f.length() - 1;
+        start = (int)f.length() - 1;
     }
 
-    int i = f.rfind('/', start);
-    int j = f.rfind('\\', start);
-    return max(i, j);
+    std::size_t i = f.rfind('/', (std::size_t)start);
+    std::size_t j = f.rfind('\\', (std::size_t)start);
+    return (int)max(i, j);
 }
 
 /**
