--- G3D.lib/include/G3D/System.h	2010-01-27 12:19:49.000000000 -0500
+++ /Users/mixamolaptop3/Downloads/G3D/G3D.lib/include/G3D/System.h	2012-09-14 03:11:31.000000000 -0400
@@ -440,20 +440,31 @@
 
 
 #ifdef _MSC_VER
-    inline uint64 System::getCycleCount() {
-        uint32 timehi, timelo;
-
-        // Use the assembly instruction rdtsc, which gets the current
-        // cycle count (since the process started) and puts it in edx:eax.
-        __asm
-            {
-                rdtsc;
-                mov timehi, edx;
-                mov timelo, eax;
-            }
-
-        return ((uint64)timehi << 32) + (uint64)timelo;
-    }
+#   ifdef _M_X64
+        // 64-bit
+        inline uint64 System::getCycleCount() {
+            LARGE_INTEGER now;
+            QueryPerformanceCounter(&now);
+            return now.QuadPart;
+        }
+#   else
+        // 32-bit
+        inline uint64 System::getCycleCount() {
+            uint32 timehi, timelo;
+
+            // Use the assembly instruction rdtsc, which gets the current
+            // cycle count (since the process started) and puts it in edx:eax.
+            __asm
+                {
+                    rdtsc;
+                    mov timehi, edx;
+                    mov timelo, eax;
+                }
+
+            return ((uint64)timehi << 32) + (uint64)timelo;
+        }
+
+#   endif
 
 #elif defined(G3D_LINUX)
 
