--- G3D.lib/include/G3D/Vector2.h	2010-01-27 12:19:49.000000000 -0500
+++ /Volumes/BOOTCAMP/Users/sid/Documents/GitHub/Fuse_OldWin/Code/Dependencies/Thea/Code/Dependencies/G3D/G3D//G3D.lib/include/G3D/Vector2.h	2012-08-18 01:34:15.000000000 -0400
@@ -159,7 +159,7 @@
      Make this vector have unit length and return the old length.
      If the vector length was less than tolerance, do not normalize.
      */
-    float unitize(float fTolerance = 1e-06);
+    float unitize(float fTolerance = 1e-30);
 
     Vector2 min(const Vector2& v) const;
     Vector2 max(const Vector2& v) const;
