--- G3D.lib/include/G3D/Vector3.h	2010-02-26 11:18:00.000000000 -0500
+++ /Volumes/BOOTCAMP/Users/sid/Documents/GitHub/Fuse_OldWin/Code/Dependencies/Thea/Code/Dependencies/G3D/G3D//G3D.lib/include/G3D/Vector3.h	2012-08-18 01:34:15.000000000 -0400
@@ -268,7 +268,7 @@
 	
     float __fastcall dot(const Vector3& rkVector) const;
     
-    float unitize(float tolerance = 1e-06);
+    float unitize(float tolerance = 1e-30);
 
     /** Cross product.  Note that two cross products in a row
         can be computed more cheaply: v1 x (v2 x v3) = (v1 dot v3) v2  - (v1 dot v2) v3.
