--- G3D.lib/include/G3D/XML.h	2010-02-26 11:33:46.000000000 -0500
+++ /Users/mixamolaptop3/Downloads/G3D/G3D.lib/include/G3D/XML.h	2012-09-14 03:11:31.000000000 -0400
@@ -128,7 +128,7 @@
 
     /** Attribute table size; zero for a TAG */
     int numAttributes() const {
-        return m_attribute.size();
+        return (int)m_attribute.size();
     }
 
     /** Return child \a i.  Children are nested tags and the unquoted
