*** buildg3d.old	2009-02-11 11:13:30.000000000 -0800
--- buildg3d	2009-02-11 11:13:34.000000000 -0800
***************
*** 237,247 ****
  
  def libTarget():
       x = 0
  
       if windows:
!          x = VC9('VC9/G3D.sln', ['jpeg.lib', 'png.lib', 'zlib.lib', 'zip.lib', 'G3D.lib', 'GLG3D.lib']);
               
       else:
      
           if osx:
               # Build jpeg
--- 237,248 ----
  
  def libTarget():
       x = 0
  
       if windows:
!          # x = VC9('VC9/G3D.sln', ['jpeg.lib', 'png.lib', 'zlib.lib', 'zip.lib', 'G3D.lib', 'GLG3D.lib']);
!          x = VC9('VC9/G3D.sln', ['jpeg.lib', 'png.lib', 'zlib.lib', 'zip.lib', 'G3D.lib']);
               
       else:
      
           if osx:
               # Build jpeg
***************
*** 251,265 ****
               # Build png
               if x == 0: x = runicompile('png.lib')
  
           if x == 0: x = runicompile('zip.lib')
           
!          if x == 0: x = buildNativeFFMPEG()
  
           if x == 0: x = runicompile('G3D.lib')
  
!          if x == 0: x = runicompile('GLG3D.lib')
  
           
       return x
  
  ##################################################################################
--- 252,266 ----
               # Build png
               if x == 0: x = runicompile('png.lib')
  
           if x == 0: x = runicompile('zip.lib')
           
!          # if x == 0: x = buildNativeFFMPEG()
  
           if x == 0: x = runicompile('G3D.lib')
  
!          # if x == 0: x = runicompile('GLG3D.lib')
  
           
       return x
  
  ##################################################################################
