--- tmp/G3D/bin/icompile	2010-03-08 21:55:35.000000000 -0800
+++ G3D/bin/icompile	2012-07-31 13:13:32.000000000 -0700
@@ -835,12 +835,12 @@
                 state.detectedTuning = 'i686';
                 
                 # To get 64-bit as well (commented out because G3D OS X doesn't support this)
-                #arch += ['-arch', 'x86_64', '-msse2', '-msse3']
+                arch += ['-arch', 'x86_64', '-msse2', '-msse3']
         else:
             state.detectedArchitecture = 'x86_64';
                 
-        if (state.universalBinary) or (machine() != 'i386'):
-            arch += ['-arch', 'ppc']
+        # if (state.universalBinary) or (machine() != 'i386'):
+        #     arch += ['-arch', 'ppc']
 
         state.compilerOptions += arch
         state.linkerOptions   += arch
